### PR TITLE
Set resources for dataloading, training and prediction

### LINKED ISSUE #44: Investigate cpu usage in training
From #27,  job on small subset used only ~37% of the available CPU resources, the task may not scale efficiently to 256 cores, room for optimization and investigation how the model uses cpus.

### PR DESCRIPTION
**Pull request by SarahAlidoost**
closes #44

This PR:
- adds a util function to set the resources and shows how to set them in the example notebook
- fixes a few minor things
- adds a comparison between baseline (monthly average) and model results on residuals via plots

### CODE REVIEW

**Comment by SarahAlidoost** [comment]
@meiertgrootes and @rogerkuou when working on this PR, I found that the land mask from era5 does not mask the large water bodies, for example see the plots below, the white pixels are related to Lake Victoria in Africa. These large bodies are not available in "averaged" daily sst data. Something to be discussed in our meeting (data checkup). 

climanet_landmask_1

**Blue regions are water (sea, lakes, rivers,...), green are land.**

climanet_landmask_2

**White pixels (in land region) are nan in "averaged" daily sst data.**

**Comment by SarahAlidoost** [comment]
@meiertgrootes  the notebook uses residuals as target and compares the model output with the baseline (averaged daily data).

File: climanet/dataset.py [hunk 1]
```diff
@@ -148,19 +148,19 @@ def __getitem__(self, idx):
...
             land_patch = self.land_mask_np[i : i + ph, j : j + pw]  # (H, W)
-            land_tensor = torch.from_numpy(land_patch.copy()).bool()
+            land_tensor = torch.from_numpy(np.ascontiguousarray(land_patch)).bool()
         else:
             land_tensor = torch.zeros(ph, pw, dtype=torch.bool)
 
         # Convert to tensors (from_numpy is zero-copy on contiguous arrays)
         # (1, M, T, H, W)
-        daily_tensor = torch.from_numpy(daily_patch).float().unsqueeze(0)
+        daily_tensor = torch.from_numpy(daily_patch).unsqueeze(0)
```
Comment (meiertgrootes) [inline climanet/dataset.py:228]: Given the choice to force contiguous arrays above. Is it certain that daily and monthly patch are contiguous at this point? Should that be enforced?

File: climanet/train.py
Comment (meiertgrootes) [inline climanet/train.py:1]: The modularization is a very nice improvement. Use of multiple loader workers is a clear benefit.
Tying pinned memory to use of cuda should also improve performance overall.

File: climanet/utils.py
Comment (meiertgrootes) [inline climanet/utils.py:1]: The visualization and the configuration of computational resources are both very nice to have! Histograms of the target residuals the predicted corrections and the error after correction might also be quite informative in judging and demonstrating performance.

**Review by meiertgrootes — APPROVED** [review]
The PR looks good! All changes should be very useful. I've made acomment suggesting possible additional diagnostic plots. Feel free to relegate those to a separate issue if that works better.

File: climanet/utils.py
Comment (SarahAlidoost) [reply climanet/utils.py:1]: yes, that's right. I'll merge two notebooks, main and this branch to have both plots.

File: climanet/dataset.py (hunk 1 above)
Comment (SarahAlidoost) [reply climanet/dataset.py:228]: good point! since in `init` method we apply `.copy().astype(np.float32)` on them, they are contiguous arrays.
